scripts/gb_model/process_heat_demand_shape.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-FileCopyrightText: gb-open-market-model contributors
+#
+# SPDX-License-Identifier: MIT
+
+"""
+heat demand shape profile generator.
+
+This script generates the heat demand shape for each sector.
+"""
+
+import logging
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from scripts._helpers import configure_logging, set_scenario_config
+
+logger = logging.getLogger(__name__)
+
+
+def process_demand_timeseries(
+    load_path: str, cop_profile_path: str, heating_mix_path: str, sector: str
+) -> pd.DataFrame:
+    """
+    Generate load profile shape for heat pump demands in each sector
+
+    Args:
+        load_path(str): Filepath to the hourly heat load profiles generated by PyPSA-Eur
+        cop_profile_path(str): Filepath to hourly weighted COP profiles generated by PyPSA-Eur
+        heating_mix_path(str): Filepath to heating technology mix for each heating sector in FES
+        sector(str): Sector considered (residential/services)
+
+    Returns:
+        pd.DataFrame : electrified heat demand profile for the sector
+    """
+
+    cop_profile = pd.read_csv(cop_profile_path, index_col=[0, 1], parse_dates=["time"])
+    heating_mix = pd.read_csv(heating_mix_path, index_col=0)
+
+    # Read the electricity demand base .nc file
+    load = xr.open_dataset(load_path).to_dataframe().squeeze(axis=1).unstack("node")
+
+    system = ["space", "water"]
+    nodes = cop_profile.index.get_level_values(1).unique()
+    load_profile = pd.DataFrame(index=load.index, columns=nodes)
+
+    load_profile[sector] = np.zeros(len(load.index))
+    for sys in system:
+        if sector == "commercial":
+            sector_key = "services"
+        else:
+            sector_key = sector
+        load_sys = load[f"{sector_key} {sys}"]
+
+        # Normalized the load profile
+        load_normalized = load_sys / load_sys.sum()
+
+        # Compute the electrified heat demand by dividing the heat load by COP
+        pump_load = (
+            load_normalized.div(cop_profile["ASHP"].unstack("name"))
+            * heating_mix.loc["ASHP", sector]
+            + load_normalized.div(cop_profile["GSHP"].unstack("name"))
+            * heating_mix.loc["GSHP", sector]
+            + load_normalized * heating_mix.loc["Electric resistive", sector]
+        )
+
+        pump_load.replace([np.inf, -np.inf], 0, inplace=True)
```

Comment:
when would you expect to get inf values? Is this indicative of a bug somewhere else?

Reply:
ok. This is probably fine if we move the load shape normalisation to the end of the script. My concern is that zero/inf values might skew the resulting shape. Perhaps here add a check here for any inf/nan values and then raise a warning if there are, then do the filling.